### ISSUE TITLE
discrepancy between Matlab and scipy with same parameters

### ISSUE THREAD

**Issue by mkl34** [scipy/scipy#13671]
I have some Matlab results that I'm trying to reproduce with `scipy.signal.coherence`, but I'm not getting the same results
For example,

Matlab

```
x = [1,3,5,7,4,6,7,8,9,3,5,6]
y = [3,4,7,8,9,1,2,3,4,1,5,6]

[cxy,w] = mscohere(x,y,hann(6),3,6,1)

cxy =
    0.7489
    0.6034
    0.2813
    0.3319

w =
         0
    0.1667
    0.3333
    0.5000
```

Python

```
from scipy.signal import coherence
x = [1,3,5,7,4,6,7,8,9,3,5,6]
y = [3,4,7,8,9,1,2,3,4,1,5,6]

coherence(x,y,'hann',noverlap=3,nperseg=6,fs=1,detrend=False)

(array([0.        , 0.16666667, 0.33333333, 0.5       ]),
 array([0.76762535, 0.53185638, 0.32743784, 0.07759385]))
```

I believe the parameters fed to `scipy.signal.coherence` are the same as those fed to the `mscohere `function, so why am I not getting identical results?

**Comment by LSturtew** [comment]
There seems to be an issue in how the hanning window is calculated.

The hanning window should be symmetrical, but sym is set to False here: https://github.com/scipy/scipy/blob/5ab7426247900db9de856e790b8bea1bd71aec49/scipy/signal/windows/windows.py#L2124

I think this is a bug that should be fixed.

**Comment by qbarthelemy** [comment]
I agree with @LSturtew : default windows for `coherence`, `welch` and `csd` should not be non-symmetric.

@mkl34 , equivalence with Matlab is obtained using a symmetric window: 
```
coherence(x, y, window=np.hanning(6), noverlap=3, nperseg=6, fs=1, detrend=False)
```
gives
```
(array([0.        , 0.16666667, 0.33333333, 0.5       ]),
 array([0.74885959, 0.6034367 , 0.28129106, 0.33187978]))
```

**Comment by psambit9791** [comment]
I am encountering a discrepancy while working with periodograms which seems to be originating from the same issue. If no one is working on it, I would like to give this a try.

**Comment by DietBru** [comment]
Changing the default behavior of an established function will break the a lot of people's code.
PR #23518 allows now the suffixes `'_periodic'` and `'_symmetric'` in the window names to make these distinctions more apparent.  The signature of `coherence` has been updated to:
```python
def coherence(x, y, fs=1.0, window='hann_periodic', nperseg=None, noverlap=None,
               nfft=None, detrend='constant', axis=-1):
```
Consult PR #23518 for further details. Any review comments are of course very welcome.
If there are no protests, I would close this issue, if PR #23518 gets merged.
